static/app/views/dashboards/widgetCard/autoSizedText.tsx

```diff
@@ -0,0 +1,143 @@
+import {useLayoutEffect, useRef} from 'react';
+import styled from '@emotion/styled';
+import * as Sentry from '@sentry/react';
+
+interface Props {
+  children: React.ReactNode;
+  maximumDifference?: number;
+}
+
+export function AutoSizedText({
+  children,
+  maximumDifference = DEFAULT_MAXIMUM_DIFFERENCE,
+}: Props) {
+  const childRef = useRef<HTMLDivElement>(null);
+
+  const fontSize = useRef<number>(0);
+  const fontSizeLowerBound = useRef<number>(0);
+  const fontSizeUpperBound = useRef<number>(0);
```

Comment:
Nit, but with multiple refs, it can more easily happen that you end up not updating some values, especially since they seem related and I see on L42,L43 you assign them as the same time. 

I would have probably opted for a single ref to store the state
```suggestion
  const font = useRef<{current: 0, range: [0, 0]}>({current: 0, range: [0, 0]}); // or separate fields for lower/upper, whatever you prefer 
  
  // this then becomes a single line assignment
  font.current.range = [min, max];
```

Reply:
I see what you're saying! Conversely though, I also sometimes update just the lower bound and the font, or just the upper bound and the font. Grouping the bounds together didn't feel great either